pyerrors/fits.py

```diff
@@ -151,6 +151,12 @@ def func_b(a, x):
         For details about how the covariance matrix is estimated see `pyerrors.obs.covariance`.
         In practice the correlation matrix is Cholesky decomposed and inverted (instead of the covariance matrix).
         This procedure should be numerically more stable as the correlation matrix is typically better conditioned (Jacobi preconditioning).
+    corr_matrix: array, optional
+        if correlated_fit=True is set as well, can provide a correlation matrix (without y errros!) of your own choosing for a correlated fit
+    inv_chol_cov_matrix
```

Comment:
This piece is a bit tricky. In principle, the code currently expects a matrix that is based on the algorithm which is performed inside our routine. It is very easy to do something wrong, when you try to come up with the correct inverted Cholesky matrix without copying what is done in the code.

One could then ask the question whether this parameter is needed. It could be helpful when performing many fits to the same set of data because it saves some time, when the inversion would not have to be done every time. However, providing the covariance matrix could already help a lot in reducing the cost and the inversion alone is probably not too expensive. 

I did think that it would be helpful to provide the inverse, but now I cannot really come up with a good example where this would be the case. Do you have something in mind?

Reply:
Originally I thought it would be good to have one option where you are completely free to hand over whatever type of inverse covariance matrix you want. But as the code relies on the Cholesky decomposition to access the 'square root' of the chi squared function (in `general_chisqfun()`) which is cheaper and is later also needed to define the chisqfunc_residuals function I thought it would be good to keep this form of the matrix. 
I also didn't see a nice cheap way to access this if the inverse covariance matrix is handed over as whole and then only the full chi squared function is evaluated. So I essentially did not see a nice way of defining the `general_chisqfun()` function (l. 331). Do you know a nice way around that?
What I had in mind was your case above of using modified weights for different data points (see above).